### DISCUSSION TITLE
how can we add a new or more than one authorized layer to this package?

### DISCUSSION THREAD

**Discussion by m-jooibar** [spatie/laravel-permission#2408] [Q&A]
Hello . 
I want to design a system that in that a super admin can declare roles layers dynamically.
for example a super admin can add : company layer , then he will add department layer and then add role and permission layer. 
each user can be member of one or more than one companies and multiple department.
now i want to see does a user has authorize from company by id 3 and department id 6 and role 2 and permission 9 ?
so the code should be like it : $permission = 'company.depatrment.writer.create.1';
  how can i handle this scenario by this package?
thank you.

**Comment by parallels999** [comment]
[basic-usage/teams-permissions](https://spatie.be/docs/laravel-permission/v5/basic-usage/teams-permissions)
